### PR TITLE
docs: Pending review comments from createReducer()

### LINKED ISSUE #1884: Pending review comments from createReducer() function
## I would be willing to submit a PR for the docs :heart:

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

### PR DESCRIPTION
**Pull request by santoshyadav198613**
Fixes #1884

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by santoshyadav198613** [comment]
Hi @alex-okrushko ,
This Pr has all the pending review, please verify.

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md [hunk 1]
```diff
@@ -50,10 +50,10 @@ To fix the above violation, a new reference to the state has to be created:
 
 ```ts
 export const reducer = createReducer(initialState,
-  on(addTodo, state => ({
+  on(addTodo, (state, { payload } => ({
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]: Same here

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 2]
```diff
@@ -137,41 +137,42 @@ export const initialState: State = adapter.getInitialState({
...
-  on(UserActions.addUser, state => {
-   return adapter.addOne(action.user, state)
+export const reducer = createReducer(
+  initialState,
+  on(UserActions.addUser, (state, { user }) => {
+    return adapter.addOne(user, state)
   }),
-  on(UserActions.upsertUser, state => {
-    return adapter.upsertOne(action.user, state);
+  on(UserActions.upsertUser, (state, { user }) => {
```
[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/entity/adapter.md]: So it will be something line UserActions.payload right, and then payload.user.

[the remark this answers is not in the document]
Comment (alex-okrushko) [reply projects/ngrx.io/content/guide/entity/adapter.md]: > I think for the docs it would be clearer if we don't deconstruct and use action.user?

I'd rather deconstruct (because we can do that now 😃) and add the comment that it's action that is deconstructed.
Also, we frequently deconstruct in Effect, I don't think this is any different 🙂

[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/entity/adapter.md]: Ok let me know what is final, so I can make the changes. :)

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md (hunk 1 above)
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]: These should be using `todo` instead of `payload`

File: projects/ngrx.io/content/guide/entity/adapter.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-okrushko) [reply projects/ngrx.io/content/guide/entity/adapter.md]: Let's do deconstruction with the comments above it (in a few places). I feel like the code in the examples should be as real-life as possible. 😃
@brandonroberts agrees as well.

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md (hunk 1 above)
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]: Done.

File: projects/ngrx.io/content/guide/entity/adapter.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/entity/adapter.md]: Done.

**Comment by santoshyadav198613** [comment]
Open for review.

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 3]
```diff
@@ -137,41 +137,42 @@ export const initialState: State = adapter.getInitialState({
...
+  on(UserActions.deleteUsersByPredicate, (state, { predicate }) => {
+    return adapter.removeMany(predicate, state);
   }),
-  on(UserActions.loadUsers, state => {
-    return adapter.addAll(action.users, state);
+  on(UserActions.loadUsers, (state, { users }) => {
+    return adapter.addAll(users, state);
   }),
-  on(UserActions.clearUsers, state => {
+  on(UserActions.clearUsers, (state, { selectedUserId }) => {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]: `selectedUserId` is not part of the action.

```suggestion
  on(UserActions.clearUsers, state => {
```

File: projects/ngrx.io/content/guide/store/configuration/runtime-checks.md [hunk 4]
```diff
@@ -50,10 +50,10 @@ To fix the above violation, a new reference to the state has to be created:
 
 ```ts
 export const reducer = createReducer(initialState,
-  on(addTodo, state => ({
+  on(addTodo, (state, { todo } => ({
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]:
```suggestion
  on(addTodo, (state, { todo }) => ({
```

Comment (timdeschryver) [reply projects/ngrx.io/content/guide/store/configuration/runtime-checks.md]: Could you also take another look at the other examples on this page, because most of them are wrong.
Where `payload` is used we need to change it, add the action as a second parameter and destruct it to have the right property. 

```ts
export const reducer = createReducer(initialState,
  on(addTodo, state => ({
    // Violation 1: we assign a new value to `todoInput` directly
    state.todoInput = '',
    // Violation 2: `push` modifies the array
    state.todos.push(action.payload)
  }))
);
```

should be

```ts
export const reducer = createReducer(initialState,
  on(addTodo, (state, { todo }) => ({
    // Violation 1: we assign a new value to `todoInput` directly
    state.todoInput = '',
    // Violation 2: `push` modifies the array
    state.todos.push(todo)
  }))
);
```

File: projects/ngrx.io/content/guide/entity/adapter.md (hunk 3 above)
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/entity/adapter.md]: done.

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 5]
```diff
@@ -125,53 +125,54 @@ import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
 import { User } from '../models/user.model';
 import * as UserActions from '../actions/user.actions';
 
-export interface State extends EntityState&lt;User&gt; {
+export interface State extends EntityState<User> {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]: Please revert these changes. We need the use the html entities because otherwise it won't render correctly.

image

**Comment by santoshyadav198613** [comment]
Changes done, kept the symbol where it is working and changed at other places,

File: projects/ngrx.io/content/guide/entity/adapter.md (hunk 5 above)
Comment (santoshyadav198613) [reply projects/ngrx.io/content/guide/entity/adapter.md]: done.

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ export function sortByName(a: User, b: User): number {
   return a.name.localeCompare(b.name);
 }
 
-export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;({
+export const adapter: EntityAdapter<User> = createEntityAdapter&lt;User&gt;({
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]:
```suggestion
export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;({
```

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 7]
```diff
@@ -130,48 +130,49 @@ export interface State extends EntityState&lt;User&gt; {
   selectedUserId: number | null;
 }
 
-export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;();
+export const adapter: EntityAdapter<User> = createEntityAdapter&lt;User&gt;();
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]:
```suggestion
export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;();
```

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 8]
```diff
@@ -130,48 +130,49 @@ export interface State extends EntityState&lt;User&gt; {
...
+  on(UserActions.deleteUsersByPredicate, (state, { predicate }) => {
+    return adapter.removeMany(predicate, state);
   }),
-  on(UserActions.loadUsers, state => {
-    return adapter.addAll(action.users, state);
+  on(UserActions.loadUsers, (state, { users }) => {
+    return adapter.addAll(users, state);
   }),
-  on(UserActions.clearUsers, state => {
+  on(UserActions.clearUsers, (state) => {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]:
```suggestion
  on(UserActions.clearUsers, state => {
```

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 9]
```diff
@@ -105,17 +105,17 @@ import { Update } from '@ngrx/entity';
...
-export const upsertUser = createAction('[User/API] Upsert User', props&lt;{ user: User }&gt;());
-export const addUsers = createAction('[User/API] Add Users', props&lt;{ user: User }&gt;());
-export const upsertUsers = createAction('[User/API] Upsert Users', props&lt;{ users: User[] }&gt;());
-export const updateUser = createAction('[User/API] Update User', props&lt;{ user: Update&lt;User&gt; }&gt;());
-export const updateUsers = createAction('[User/API] Update Users', props&lt;{ users: Update&lt;User&gt;[] }&gt;());
-export const mapUsers = createAction('[User/API] Map Users', props&lt;{ entityMap: EntityMap&lt;User&gt; }&gt;());
-export const deleteUser = createAction('[User/API] Delete User', props&lt;{ id: string }&gt;());
-export const deleteUsers = createAction('[User/API] Delete Users', props&lt;{ id: string[] }&gt;());
-export const deleteUsersByPredicate = createAction('[User/API] Delete Users By Predicate', props&lt;{ predicate: Predicate&lt;User&gt; }&gt;());
+export const loadUsers = createAction('[User/API] Load Users', props<{ users: User[] }>());
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]: Could it be that you forgot to push these changes? These should all use `&lt;` and `&gt;`

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 10]
```diff
@@ -105,17 +105,17 @@ import { Update } from '@ngrx/entity';
...
+export const addUser = createAction('[User/API] Add User', props<{ user: User }>());
+export const upsertUser = createAction('[User/API] Upsert User', props<{ user: User }>());
+export const addUsers = createAction('[User/API] Add Users', props<{ users: User }>());
+export const upsertUsers = createAction('[User/API] Upsert Users', props<{ users: User[] }>());
+export const updateUser = createAction('[User/API] Update User', props<{ user: Update&lt;User&gt; }>());
+export const updateUsers = createAction('[User/API] Update Users', props<{ users: Update&lt;User&gt;[] }>());
```
Comment (santoshyadav198613) [inline projects/ngrx.io/content/guide/entity/adapter.md]: Hi @timdeschryver ,
A question her, this code looks incorrect. I think it should be 
export const updateUsers = createAction('[User/API] Update Users', props<{ users: Update&lt;User[]&gt; }>());

**Comment by santoshyadav198613** [comment]
Conflicts are resolved now.

File: projects/ngrx.io/content/guide/entity/adapter.md [hunk 11]
```diff
@@ -106,17 +106,17 @@ import { Update } from '@ngrx/entity';
...
-export const updateUser = createAction('[User/API] Update User', props&lt;{ user: Update&lt;User&gt; }&gt;());
-export const updateUsers = createAction('[User/API] Update Users', props&lt;{ users: Update&lt;User&gt;[] }&gt;());
-export const mapUsers = createAction('[User/API] Map Users', props&lt;{ entityMap: EntityMap&lt;User&gt; }&gt;());
-export const deleteUser = createAction('[User/API] Delete User', props&lt;{ id: string }&gt;());
-export const deleteUsers = createAction('[User/API] Delete Users', props&lt;{ id: string[] }&gt;());
-export const deleteUsersByPredicate = createAction('[User/API] Delete Users By Predicate', props&lt;{ predicate: Predicate&lt;User&gt; }&gt;());
+export const loadUsers = createAction('[User/API] Load Users', props<{ users: User[] }>());
+export const addUser = createAction('[User/API] Add User', props<{ user: User }>());
+export const upsertUser = createAction('[User/API] Upsert User', props<{ user: User }>());
+export const addUsers = createAction('[User/API] Add Users', props<{ users: User }>());
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/entity/adapter.md]:
```suggestion
export const addUsers = createAction('[User/API] Add Users', props<{ users: User[] }>());
```

File: projects/ngrx.io/content/guide/entity/adapter.md (hunk 10 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/entity/adapter.md]: It's fine how it is currently. The signatureis  `Update<T>[]`
